gateleen-routing/src/main/java/org/swisspush/gateleen/routing/RouterConfigurationParser.java

```diff
@@ -0,0 +1,113 @@
+package org.swisspush.gateleen.routing;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.oauth2.OAuth2FlowType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.swisspush.gateleen.core.util.StringUtils;
+import org.swisspush.gateleen.routing.auth.OAuthConfiguration;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RouterConfigurationParser {
+
+    private static final Logger log = LoggerFactory.getLogger(RouterConfigurationParser.class);
+
+    private static final String REQUEST_HOPS_LIMIT_PROPERTY = "request.hops.limit";
+    private static final String AUTHCONFIGS_PROPERTY = "authConfigs";
+    private static final String FLOWTYPE_PROPERTY = "flowType";
+    private static final String CLIENTID_PROPERTY = "clientId";
+    private static final String CLIENTSECRET_PROPERTY = "clientSecret";
+    private static final String SITE_PROPERTY = "site";
+    private static final String TOKENPATH_PROPERTY = "tokenPath";
+    private static final String AUTHPATH_PROPERTY = "authPath";
+    private static final String SCOPES_PROPERTY = "scopes";
+    private static final String SUPPORTEDGRANTTYPES_PROPERTY = "supportedGrantTypes";
+
+    /**
+     * Parses the provided router configuration resource and returns a {@link RouterConfiguration}.
+     *
+     * @param configurationResourceBuffer the resource to parse
+     * @return a {@link RouterConfiguration}
+     */
+    public static Optional<RouterConfiguration> parse(Buffer configurationResourceBuffer,
+                                                      Map<String, Object> properties) {
+        String replacedConfig;
+        JsonObject config;
+
+        try {
+            replacedConfig = StringUtils.replaceWildcardConfigs(configurationResourceBuffer.toString(UTF_8), properties);
+            config = new JsonObject(Buffer.buffer(replacedConfig));
+        } catch (Exception e) {
+            log.warn("Could not replace wildcards with environment properties for the router configuration " +
+                            "due to following reason: {}", e.getMessage());
+            return Optional.empty();
```

Comment:
Is this a bit dangerous? I mean, the security won't kick if there are configuration issues and we don't easily see it? We maybe have the option to let the requests fail in such cases.

Reply:
When there is an issue with the router configuration resource, we won't have any configurations to be used. When a routing rule with a configured _oAuthId_ is used, the request will fail because no configuration can be found. Also when an endpoint that is protected with OAuth is called without authorization, the request will be rejected anyways.

Therefore, I think the current implementation is OK. What do you think?